databricks/koalas/indexes.py

```diff
@@ -390,6 +390,35 @@ def fillna(self, value):
         result = DataFrame(self._kdf._internal.copy(sdf=sdf)).index
         return result
 
+    # TODO: ADD keep parameter
+    def drop_duplicates(self):
+        """
+        Return Index with duplicate values removed.
+
+        Returns
+        -------
+        deduplicated : Index
+
+        See Also
+        --------
+        Series.drop_duplicates : Equivalent method on Series.
+        DataFrame.drop_duplicates : Equivalent method on DataFrame.
+
+        Examples
+        --------
+        Generate an pandas.Index with duplicate values.
+
+        >>> idx = pd.Index(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'])
+
+        >>> idx.drop_duplicates()
+        Index(['lama', 'cow', 'beetle', 'hippo'], dtype='object')
+        """
+        kdf = self._kdf.copy()
+        sdf = kdf._internal.sdf.select(self._internal.index_scols).drop_duplicates()
+        internal = _InternalFrame(sdf=sdf, index_map=self._internal.index_map)
+        kdf = DataFrame(internal)
+        return Index(kdf) if type(self) == Index else MultiIndex(kdf)
```

Comment:
maybe i think you can simply `return DataFrame(internal).index`

Reply:
and i think we should keep their name.

for example.

```python
>>> idx = pd.Index(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'], name='rain')
>>> idx.drop_duplicates()
Index(['lama', 'cow', 'beetle', 'hippo'], dtype='object', name='rain')
```

Reply:
Thanks your fast and kind reply. Fixing it in next commit.